scripts/core/editor3/components/Editor3.jsx

```diff
@@ -94,12 +94,17 @@ export class Editor3Component extends React.Component {
     /**
      * @ngdoc method
      * @name Editor3#onDragOver
-     * @returns {Boolean} Returns true if the item is permitted.
+     * @returns {Boolean} Returns true if the item is not permitted.
      * @description Checks if the dragged over item is allowed.
      */
     onDragOver(e) {
+        const {editorFormat, readOnly, singleLine} = this.props;
         const mediaType = e.originalEvent.dataTransfer.types[0] || '';
 
+        if (readOnly || singleLine || !(editorFormat && editorFormat.indexOf('picture') !== -1)) {
```

Comment:
This should suffice:
```js
    onDragOver(e) {
        const {editorFormat, readOnly, singleLine} = this.props;
        const mediaType = e.originalEvent.dataTransfer.types[0] || '';

        return !readOnly && editorFormat.indexOf('picture') > -1 && [
            'application/superdesk.item.picture',
            'application/superdesk.item.graphic',
            'application/superdesk.item.video',
            'text/html',
        ].indexOf(mediaType) === -1;
    }
```
Reason being that `singleLine` editors don't have formatting options anyway, so it's not needed to check that. It's also not needed to validate that `editorFormat` is non-null if you add this at the bottom:
```js
Editor3Component.defaultProps = {
    editorFormat: []
};
```
I would also write a test for this, where you call `onDragOver` directly and assert whether it returns `true` or `false` based on different props.